### ISSUE TITLE
Ask permission before deleting mode-specific files.

### ISSUE THREAD

**Issue by nothingismagick** [quasarframework/quasar#4748]
**Is your feature request related to a problem? Please describe.**
When removing a mode with e.g. `quasar mode remove electron` all of the files in that folder are destroyed and this is irrecoverable.

**Describe the solution you'd like**
I want to be asked first if I really want this to happen, and would prefer to keep any files that I have personally modified. 

**Describe alternatives you've considered**
An alternative - but really just a workaround and actually treating the symptom not the problem, would be to have a setting during install that enables git integration and automatically adds **SOME** files that quasar creates with `git add` (except of course things in the `.gitignore` context). 

**Additional context**
This accidentally happened to me the other night while developing on core. It was late and I had a command in my bash history `quasar mode remove proton && quasar mode add proton` and without really thinking about its repurcussions I purged all of my work. In discussion with @webnoob we thought that this would be something to look into.

**Comment by nothingismagick** [comment]
Note for @rstoenescu - this is a tracking issue and once we've discussed an approach I will look into making the appropriate modifications asking permission (like how AE re-install process works).

**Comment by nothingismagick** [comment]
I think deleting stuff should be opt-in - as should tampering with quasar.conf.js
Quasar does not always notify them what is going on - and that can be confusing.

Especially if you think about someone who is used to current behaviour and then we start tampering with their `quasar.conf.js` and they don't notice and then they add the mode back and its back to defaults and they're like WTF.

This is one of the things that I am especially conscious of - retaining current behavior wherever possible and making new behavior either clear with a new flag:

```
quasar mode remove proton --keep-files
quasar mode add proton --enable-tampering
```

or dev interaction:
```
$ quasar mode add proton
> Proton Object not found in quasar.conf.js. Create? Y / n
```

I am not saying we have to do one or the other at the expense of another option, just suggesting an approach that does not break the contract with the current users

**Comment by webnoob** [comment]
Another issue which is semi related to this is when we're adding a mode `quasar mode add electron` it copies the files over from the template folder but at that point we have no way of updating those files if there is a bug as it will only the files if they don't exist.

BEX handles this by always copying files on each dev and instructing the users never to edit them. They in turn have hooks the user can use to add bits in where required. 

This might be a discussion for another topic though but thought it worth mentioning in case we do this all in one.

**Comment by lucasfernog** [comment]
@webnoob that's why code on the src-{mode} folder are meant to be only the minimum necessary code to run the application, and every config related to Quasar must be done outside it e.g. @quasar/app (a good example of it is the SSR mode, its template just extends q-app functionality).

**Comment by rstoenescu** [comment]
User will be prompted before actually removing the mode.
Prompt will be auto-skipped if "--yes"/"-y" param is specified.

Target: quasar/app 1.0.5.

**Comment by webnoob** [comment]
@lucasfernog Not sure how that answers the issue I raised:

> but at that point we have no way of updating those files if there is a bug as it will only the files if they don't exist.

If we need to add something to one of those src files to support a new feature, we can't as the files are written once and only once.

**Comment by nothingismagick** [comment]
The solution provided by the MR only stops the removal. I think Allan is
right, it doesn't really resolve the situation. Because you can't really
remove the mode and keep the files. Am I wrong?

**Comment by lucasfernog** [comment]
@webnoob our code in src-mode should only call our functions on q-app or similar side of things, so no bug should be there.

**Comment by lucasfernog** [comment]
@nothingismagick removing the mode IS removing the files, that doesn't make sense if I understood it correctly.

**Comment by nothingismagick** [comment]
Maybe you are right, but my point is that the mode is not the same as the
files. If I modify the contents of the files related to the mode, I have
invested in them. Maybe I want to remove the mode and keep my work. That is
what this issue was about from the beginning.

**Comment by webnoob** [comment]
So my example would be this file: https://github.com/quasarframework/quasar/blob/dev/app/templates/electron/main-process/electron-main.js

It's common (as I understand it) to need to add code to this file for an electron app. I know I had to for the small test app I made for Icon Genie. Whilst I agree it should be bug free, we can't guarantee that and we can't guarantee we won't need to add anything in the future.

I've solved this with BEXs by always overwriting that file on build but providing "hook" files which the user can edit (that have access to the context) but don't get overwritten. See the call for `attachActivatedBackgroundHooks` as an example here: https://github.com/webnoob/quasar/blob/mode-browser-extension/app/templates/bex/js/core/background/background.js

For me, this offers the best of both worlds as it keeps our Quasar code separate from user code and always gives us an update path.
